File: LogisimPlus/LogisimPlus/logisim/gui/main/ToolboxToolbarModel.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

using System.Collections.Generic;

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.main
{

	using AbstractToolbarModel = draw.toolbar.AbstractToolbarModel;
	using ToolbarItem = draw.toolbar.ToolbarItem;
	using LogisimMenuBar = logisim.gui.menu.LogisimMenuBar;
	using logisim.util;

	internal class ToolboxToolbarModel : AbstractToolbarModel, MenuListener.EnabledListener
	{
		private LogisimToolbarItem itemAdd;
		private LogisimToolbarItem itemUp;
		private LogisimToolbarItem itemDown;
		private LogisimToolbarItem itemDelete;
		private List<ToolbarItem> items;

		public ToolboxToolbarModel(MenuListener menu)
		{
			itemAdd = new LogisimToolbarItem(menu, "projadd.gif", LogisimMenuBar.ADD_CIRCUIT, Strings.getter("projectAddCircuitTip"));
			itemUp = new LogisimToolbarItem(menu, "projup.gif", LogisimMenuBar.MOVE_CIRCUIT_UP, Strings.getter("projectMoveCircuitUpTip"));
			itemDown = new LogisimToolbarItem(menu, "projdown.gif", LogisimMenuBar.MOVE_CIRCUIT_DOWN, Strings.getter("projectMoveCircuitDownTip"));
			itemDelete = new LogisimToolbarItem(menu, "projdel.gif", LogisimMenuBar.REMOVE_CIRCUIT, Strings.getter("projectRemoveCircuitTip"));

			items = UnmodifiableList.create(new ToolbarItem[] {itemAdd, itemUp, itemDown, itemDelete});

			menu.addEnabledListener(this);
		}

		public override List<ToolbarItem> Items
		{
			get
			{
				return items;
			}
		}

		public override bool isSelected(ToolbarItem item)
		{
			return false;
		}

		public override void itemSelected(ToolbarItem item)
		{
			if (item is LogisimToolbarItem)
			{
				((LogisimToolbarItem) item).doAction();
			}
		}

		//
		// EnabledListener methods
		//
		public virtual void menuEnableChanged(MenuListener source)
		{
			fireToolbarAppearanceChanged();
		}
	}

}

File: logisim/app/src/main/java/logisim/gui/main/ToolboxToolbarModel.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.main;

import java.util.List;

import draw.toolbar.AbstractToolbarModel;
import draw.toolbar.ToolbarItem;
import logisim.gui.main.MenuListener.EnabledListener;
import logisim.gui.menu.LogisimMenuBar;
import logisim.util.UnmodifiableList;

class ToolboxToolbarModel extends AbstractToolbarModel implements EnabledListener {
	private List<ToolbarItem> items;

	public ToolboxToolbarModel(MenuListener menu) {
		LogisimToolbarItem itemAdd = new LogisimToolbarItem(menu, "projadd.gif", LogisimMenuBar.ADD_CIRCUIT,
				Strings.getter("projectAddCircuitTip"));
		LogisimToolbarItem itemUp = new LogisimToolbarItem(menu, "projup.gif", LogisimMenuBar.MOVE_CIRCUIT_UP,
				Strings.getter("projectMoveCircuitUpTip"));
		LogisimToolbarItem itemDown = new LogisimToolbarItem(menu, "projdown.gif", LogisimMenuBar.MOVE_CIRCUIT_DOWN,
				Strings.getter("projectMoveCircuitDownTip"));
		LogisimToolbarItem itemDelete = new LogisimToolbarItem(menu, "projdel.gif", LogisimMenuBar.REMOVE_CIRCUIT,
				Strings.getter("projectRemoveCircuitTip"));

		items = UnmodifiableList.create(new ToolbarItem[] {itemAdd, itemUp, itemDown, itemDelete, });

		menu.addEnabledListener(this);
	}

	@Override
	public List<ToolbarItem> getItems() {
		return items;
	}

	@Override
	public boolean isSelected(ToolbarItem item) {
		return false;
	}

	@Override
	public void itemSelected(ToolbarItem item) {
		if (item instanceof LogisimToolbarItem) ((LogisimToolbarItem) item).doAction();
	}

	//
	// EnabledListener methods
	//
	public void menuEnableChanged(MenuListener source) {
		fireToolbarAppearanceChanged();
	}
}
